kernel/src/grant.rs

```diff
@@ -3,29 +3,26 @@
 use core::marker::PhantomData;
 use core::mem::{align_of, size_of};
 use core::ops::{Deref, DerefMut};
-use core::ptr::{slice_from_raw_parts_mut, write, NonNull};
+use core::ptr::{write, NonNull};
 
-use crate::process::{Error, ProcessType};
+use crate::process::{Error, ProcessDynamicGrantIdentifer, ProcessType};
 use crate::sched::Kernel;
 use crate::upcall::AppId;
 
-/// Type that indicates a grant region has been entered and borrowed.
-/// This is passed to capsules when they try to enter a grant region.
+/// This type provides access to the memory in a process's grant.
+///
+/// The grant must be entered to retrieve a `Borrowed` object. This object is
```

Comment:
It seems like this should describe the type, rather than how you obtain it? Something like "A `Borrowed` object provides access to memory with a process's grant region. Borrow objects are typed by what a particular grant stores. Kernel code gains access to a `Borrowed` object by calling `Grant::enter`."